clippy_lints/src/utils/sugg.rs

```diff
@@ -263,21 +304,53 @@ impl<'a> Sugg<'a> {
     }
 }
 
-impl<'a, 'b> std::ops::Add<Sugg<'b>> for Sugg<'a> {
+// Copied from the rust standart library, and then edited
+macro_rules! forward_binop_impls_to_ref {
+    (impl $imp:ident, $method:ident for $t:ty, type Output = $o:ty) => {
+        impl $imp<$t> for &$t {
+            type Output = $o;
+
+            fn $method(self, other: $t) -> $o {
+                $imp::$method(self, &other)
+            }
+        }
+
+        impl $imp<&$t> for $t {
+            type Output = $o;
+
+            fn $method(self, other: &$t) -> $o {
+                $imp::$method(&self, other)
+            }
+        }
+
+        impl $imp for $t {
+            type Output = $o;
+
+            fn $method(self, other: $t) -> $o {
+                $imp::$method(&self, &other)
+            }
+        }
+    };
+}
+
+impl Add for &Sugg<'_> {
```

Comment:
Yes that macro is what I meant. Can you also do this for the other operation?

Reply:
I've implemented `Neg` for `Sugg` without the reference versions, instead of direct calls of `make_binop` for now. Just to make it clear, do you mean overloading other operations like `Mul` for  future use regardless what this PR needs or implementing forwarding for unary operators? If the answer is the latter, as `make_unop` takes ownership, `self` have to be cloned to do so; or it will probably be possible to make `make_unop` taking references; which one do you think preferable?

Reply:
Ah I see, I should have looked closer at what is implemented for `Sugg`. In that case, you can keep it, as it is right now.

Reply:
I understand, I will do so. ...I mean making no change.